src/openfl/_internal/stage3D/opengl/GLContext3D.hx

```diff
@@ -638,103 +593,82 @@ class GLContext3D {
 		
 		var gl = context.__renderSession.gl;
 		
-		var width = 0;
-		var height = 0;
+		var width = texture.__width;
+		var height = texture.__height;
+		var create = texture.__framebuffer == null;
 		
-		if (context.__framebuffer == null) {
+		if (create) {
 			
-			context.__framebuffer = gl.createFramebuffer ();
+			texture.__framebuffer = gl.createFramebuffer ();
 			GLUtils.CheckGLError ();
 			
 		}
 		
-		gl.bindFramebuffer (gl.FRAMEBUFFER, context.__framebuffer);
+		gl.bindFramebuffer (gl.FRAMEBUFFER, texture.__framebuffer);
 		GLUtils.CheckGLError ();
 		
-		if (Std.is (texture, Texture)) {
-			
-			var texture2D:Texture = cast texture;
-			width = texture2D.__width;
-			height = texture2D.__height;
-			
-			gl.framebufferTexture2D (gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture.__textureID, 0);
-			GLUtils.CheckGLError ();
-			
-		} else if (Std.is (texture, RectangleTexture)) {
-			
-			var rectTexture:RectangleTexture = cast texture;
-			width = rectTexture.__width;
-			height = rectTexture.__height;
-			
-			gl.framebufferTexture2D (gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture.__textureID, 0);
-			GLUtils.CheckGLError ();
-			
-		} else if (Std.is (texture, CubeTexture)) {
+		if (create) {
 			
-			var cubeTexture:CubeTexture = cast texture;
-			width = cubeTexture.__size;
-			height = cubeTexture.__size;
-			
-			for (i in 0...6) {
+			if (Std.is (texture, Texture)) {
+				
+				gl.framebufferTexture2D (gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture.__textureID, 0);
+				GLUtils.CheckGLError ();
+				
+			} else if (Std.is (texture, RectangleTexture)) {
```

Comment:
the code in this branch is the same, so maybe we can do`Std.is(texture,Texture) || Std.is(texture, RectangleTexture)` and avoid a bit of duplication